include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -39,6 +39,25 @@ namespace Utilities
   {
     namespace ConsensusAlgorithms
     {
+      namespace
+      {
+        /**
+         * Return whether a vector of targets (MPI ranks) has only unique
+         * elements.
+         */
+        bool
+        has_unique_elements(const std::vector<unsigned int> &targets)
+        {
+          std::vector<unsigned int> my_destinations = targets;
+          const unsigned int        n_destinations  = my_destinations.size();
+          std::sort(my_destinations.begin(), my_destinations.end());
+          my_destinations.erase(std::unique(my_destinations.begin(),
+                                            my_destinations.end()),
+                                my_destinations.end());
+          return (my_destinations.size() == n_destinations);
```

Comment:
Can't you apply the trick from the other PR,
```suggestion
          return std::adjacent_find(my_destinations.begin(),
                                            my_destinations.end()) == 
                                my_destinations.end();
```

Reply:
I can! So updated.